pkg/pipeline/transform/infinispan/cache.go

```diff
@@ -0,0 +1,90 @@
+package infinispan
+
+import (
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"regexp"
+
+	ispnclient "github.com/infinispan/infinispan-operator/pkg/http"
+	"github.com/infinispan/infinispan-operator/pkg/mime"
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	"github.com/sirupsen/logrus"
+
+	"github.com/infinispan/infinispan-operator/pkg/infinispan/client/api"
+	v14 "github.com/infinispan/infinispan-operator/pkg/infinispan/client/v14"
+)
+
+var log = logrus.WithField("component", "transform.Network.Cache")
+var regex = regexp.MustCompile(`[^a-zA-Z0-9 ]+`)
+
+const (
+	// TODO: implements other cache scenarios
+	// K8SCacheName          = "K8SCache"
+	// ConntrackCacheName    = "Conntrack"
+	// MultiClusterCacheName = "MultiCluster"
+	DedupCacheName = "deduper"
+)
+
+type Cache struct {
+	ispnclient.HttpClient
+	ispn api.Infinispan
+}
+
+func NewCache(kubeConfigPath string) (*Cache, error) {
+	log.Debug("Entering NewCache ...")
+
+	client := New(Config{
+		Endpoint: "http://cache.netobserv.svc.cluster.local:11222",
+		/*Credentials: &Credentials{
+			Username: "netobserv",
+			Password: "netobserv",
+		},*/
+	})
+	ispn := v14.New(client)
+
+	return &Cache{
+		HttpClient: client,
+		ispn:       ispn,
+	}, nil
+}
+
+func (c *Cache) getDedupKey(v config.GenericMap) string {
+	return regex.ReplaceAllString(
+		fmt.Sprintf("%d-%d-%s-%d-%s-%d", v["TimeFlowEndMs"], v["Proto"], v["SrcAddr"], v["SrcPort"], v["DstAddr"], v["DstPort"]),
```

Comment:
I put `TimeFlowEndMs` on top of the 5 tuple to avoid conflicts when 2 Egress comes before first Ingress.
As is, I miss some flows as you can see in the screens. We need to find a better way to unique identify each flows.

Any thoughts ?

Reply:
Perhaps we can do an assumption here, that the interfaces are not going to change over time across flows in the same 5-tuples. In fact I'm already doing this same assumption [here](https://github.com/netobserv/network-observability-console-plugin/pull/489/files#diff-b428443ff8eb670d5b01d47a599d4d67f24acddd9857e1dd327378499a2e3409R42) 
On top of that, it should be much better for the memory footprint.

The downside is that it works for merging interfaces, but might not work if we want to reuse the cache for other purposes.

Reply:
this cache is only for deduper so I'm fine with that; 

However are we sure that the interfaces are not changing when the pod is recreated for example ?
The ip addresses can be the same but ovn generated interfaces could change :thinking:

Reply:
on top of that, I could push only fields we want to merge in the cache instead of the whole flow

Reply:
> However are we sure that the interfaces are not changing when the pod is recreated for example ?
> The ip addresses can be the same but ovn generated interfaces could change

A pod restart would very very very likely use a different IP as far as I know; and even if that's not the case, the connection would need to be reinitiated so the source port would change

Reply:
but perhaps in any case there should be an expiry mechanism

Reply:
the cache already expire when full :smile_cat: you can also set a lifespan
https://infinispan.org/docs/stable/titles/configuring/configuring.html#expiration_configuring-memory-usage

Reply:
I made changes in https://github.com/netobserv/flowlogs-pipeline/pull/639/commits/335d7370beb104917ab9806b4663b6f40a1f42f6